lib/process.js

```diff
@@ -1,48 +1,74 @@
   // preventing us from double paying packages from an org's donation.
   // This will throw if it's locked
   const lockInfo = await dynamo.lockOrg({ organizationId })
   log.info({ lockInfo })
 
-  const { name, installationId } = await db.getOrg({ organizationId })
-
-  // get manifest search patterns for each supported registry+language
-  // e.g. "package.json" is the only manifest search pattern for JavaScript/NPM
-  // this call returns a list of [{ registry, language, patterns }, ...]
-  const searchPatterns = resolver.getSupportedManifestPatterns()
-  log.info('Using %d search pattern(s) to find package manifest files within org', searchPatterns.length)
-
-  // call the code host (e.g. GitHub) to search all the org's repos for each of the search patterns
-  // this call returns a list of [{ registry, language, manifest }, ...] -- that is, a separate
-  // object for each manifest file found, alongside its registry and language. the manifest is unparsed (raw utf8)
-  const packageManifests = await retriever.getAllManifestsForOrg({ name, installationId }, searchPatterns)
-  log.info('Downloaded %d package manifests', packageManifests.length)
-
-  // now ask the registry resolver to parse the manifest files according to whichever registry/language they are
-  // so, for example, { registry: npm, language: javascript, manifest: <some JSON string> } will be parsed as
-  // JSON and the dependencies+devDependencies fields will be extracted as top level dependencies.
-  // this call returns a list of [{ registry, language, deps }, ...] for each registry and language -- even if
-  // there are many unique manifests passed in for the registry and language. it will group all the deps for
-  // the registry/language combination into a single list.
-  const extractedDependencies = resolver.extractDependenciesFromManifests(packageManifests)
-  log.info('Dependencies extracted for %d different registry/language combinations', extractedDependencies.length)
+  // a list of { language: string, registry: string, deps: []string }
+  let extractedDependencies
+
+  if (targetPackageId) {
+    // we need the name, language, and registry information for this package
+    const pkg = await db.getPackage({ packageId: targetPackageId })
+    if (!pkg) throw new Error(`targetPackageId not found in db: ${targetPackageId}`)
+    const { name, language, registry } = pkg
+    if (!name || !language || !registry) throw new Error(`missing properties on target package: ${name} ${language} ${registry}`)
+
+    // now we need to create a string that looks like a top level package so that the
+    // registry resolver can determine this package's dependencies
+    const pkgSpec = resolver.buildLatestSpec(name, { language, registry })
+
+    // we are now ready to find this package's dep tree and disperse the dono
+    extractedDependencies = [{ language, registry, deps: [pkgSpec] }]
+  } else {
+    const { name, installationId } = await db.getOrg({ organizationId })
```

Comment:
Can you add a brief summary comment for what this else block does:

“Otherwise, if no target package id is passed in, it means we’re distributing to an entire orgs deps, so we’ll hook into GitHub to find their repos and then their deps etc etc